src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -1681,9 +1681,27 @@ internal bool ShutdownLoop
                 }
             }
 
+            /// <summary>
+            /// Call WaitForCommEvent (which is a thread function for a background thread)
+            /// within an exception handler, so that unhandled exceptions in WaitForCommEvent
+            /// don't cause process termination
+            /// </summary>
+            internal void SafelyWaitForCommEvent()
+            {
+                try
+                {
+                    WaitForCommEvent();
+                }
+                catch
+                {
+                    //TODO What is the correct behaviour for discarding exceptions
```

Comment:
But should we have a test that causes a throw out of a background thread? Verifies your change